File: src/AndroidClient/android/src/androidTest/java/net/servicestack/android/test/TestInterfaceMarkerTests.java
package net.servicestack.android.test;

import junit.framework.TestCase;

import net.servicestack.client.HttpMethods;
import net.servicestack.client.JsonServiceClient;

import static net.servicestack.android.test.dtos.*;

/**
 * Created by mythz on 9/11/2015.
 */
public class TestInterfaceMarkerTests  extends TestCase {
    public TestInterfaceMarkerTests() {

    }

    JsonServiceClient client = new JsonServiceClient("https://test.servicestack.net");

    public void test_Does_SendDefault_as_POST(){
        SendVerbResponse response = client.send(new SendDefault().setId(1));
        assertEquals(1, (int)response.getId());
        assertEquals(HttpMethods.Post, response.getRequestMethod());
        assertEquals("/json/reply/SendDefault", response.getPathInfo());
    }

    public void test_Does_SendRestGet_as_GET_using_Predefined_Route(){
        SendVerbResponse response = client.send(new SendRestGet().setId(1));
        assertEquals(1, (int)response.getId());
        assertEquals(HttpMethods.Get, response.getRequestMethod());
        assertEquals("/json/reply/SendRestGet", response.getPathInfo());
    }

    public void test_Does_SendGet_as_GET(){
        SendVerbResponse response = client.send(new SendGet().setId(1));
        assertEquals(1, (int)response.getId());
        assertEquals(HttpMethods.Get, response.getRequestMethod());
        assertEquals("/json/reply/SendGet", response.getPathInfo());
    }

    public void test_Does_SendPost_as_POST(){
        SendVerbResponse response = client.send(new SendPost().setId(1));
        assertEquals(1, (int)response.getId());
        assertEquals(HttpMethods.Post, response.getRequestMethod());
        assertEquals("/json/reply/SendPost", response.getPathInfo());
    }

    public void test_Does_SendPut_as_PUT(){
        SendVerbResponse response = client.send(new SendPut().setId(1));
        assertEquals(1, (int)response.getId());
        assertEquals(HttpMethods.Put, response.getRequestMethod());
        assertEquals("/json/reply/SendPut", response.getPathInfo());
    }
}

File: src/legacy/kotlin/src/androidTest/java/test/servicestack/net/kotlin/test/TestInterfaceMarkerTests.kt
//  Copyright (c) 2015 ServiceStack LLC. All rights reserved.

package test.servicestack.net.kotlin.test

import junit.framework.Assert
import junit.framework.TestCase
import net.servicestack.client.HttpMethods
import net.servicestack.client.JsonServiceClient

class TestInterfaceMarkerTests : TestCase() {

    internal var client = JsonServiceClient("https://test.servicestack.net")

    fun test_Does_SendDefault_as_POST() {
        val request = SendDefault()
        request.id = 1
        val response = client.send(request)
        Assert.assertEquals(1, response.id!!)
        Assert.assertEquals(HttpMethods.Post, response.requestMethod)
        Assert.assertEquals("/json/reply/SendDefault", response.pathInfo)
    }

    fun test_Does_SendRestGet_as_GET_using_Predefined_Route() {
        val request = SendRestGet()
        request.id = 1
        val response = client.send(request)
        Assert.assertEquals(1, response.id!!)
        Assert.assertEquals(HttpMethods.Get, response.requestMethod)
        Assert.assertEquals("/json/reply/SendRestGet", response.pathInfo)
    }

    fun test_Does_SendGet_as_GET() {
        val request = SendGet()
        request.id = 1
        val response = client.send(request)
        Assert.assertEquals(1, response.id!!)
        Assert.assertEquals(HttpMethods.Get, response.requestMethod)
        Assert.assertEquals("/json/reply/SendGet", response.pathInfo)
    }

    fun test_Does_SendPost_as_POST() {
        val request = SendPost()
        request.id = 1
        val response = client.send(request)
        Assert.assertEquals(1, response.id!!)
        Assert.assertEquals(HttpMethods.Post, response.requestMethod)
        Assert.assertEquals("/json/reply/SendPost", response.pathInfo)
    }

    fun test_Does_SendPut_as_PUT() {
        val request = SendPut()
        request.id = 1
        val response = client.send(request)
        Assert.assertEquals(1, response.id!!)
        Assert.assertEquals(HttpMethods.Put, response.requestMethod)
        Assert.assertEquals("/json/reply/SendPut", response.pathInfo)
    }
}
